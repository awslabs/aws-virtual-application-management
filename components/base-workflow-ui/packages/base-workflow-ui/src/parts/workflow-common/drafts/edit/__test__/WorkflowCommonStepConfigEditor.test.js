/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import WorkflowCommonStepConfigEditor from '../WorkflowCommonStepConfigEditor';

describe('WorkflowCommonStepConfigEditor', () => {
  //   const mockEvent = { preventDefault: jest.fn(), stopPropagation: jest.fn() };
  // eslint-disable-next-line no-unused-vars
  let component = null;
  let wrapper = null;
  const form = { errors: () => [] };
  const cancel = jest.fn();
  const previous = jest.fn();
  const onSave = jest.fn();
  const applyChanges = jest.fn();
  const restart = jest.fn();
  const entries = () => {
    return [['key', 'value']];
  };
  const stepEditor = {
    step: jest.fn(),
    configEdit: jest.fn(),
    setConfigEdit: jest.fn(),
    configurationEditor: {
      form,
      applyChanges,
      restart,
      previous,
      cancel,
      definedConfigList: [],
      configuration: { entries },
      review: 'review',
    },
  };
  const className = 'test-className';
  const onCancel = jest.fn();
  const stepTemplatesStore = { load: jest.fn() };
  beforeEach(() => {
    // Render component
    wrapper = shallow(
      <WorkflowCommonStepConfigEditor
        stepEditor={stepEditor}
        onSave={onSave}
        className={className}
        onCancel={onCancel}
      />,
    );

    // Get instance of the component
    component = wrapper.instance();
  });
  it('renders correctly', () =>
    expect(
      renderer
        .create(
          <Provider stepTemplatesStore={stepTemplatesStore}>
            <WorkflowCommonStepConfigEditor stepEditor={stepEditor} onSave={onSave} className={className} />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot());
});
