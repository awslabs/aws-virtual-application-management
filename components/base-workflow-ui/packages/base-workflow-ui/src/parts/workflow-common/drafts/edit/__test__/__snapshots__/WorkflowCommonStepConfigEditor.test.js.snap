// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowCommonStepConfigEditor renders correctly 1`] = `
<div
  className="test-className"
>
  <div
    className="ui divider mt1 mb2"
  />
  <h2
    className="ui grey center aligned header mt1 mb3"
  >
    Review Configuration Changes
  </h2>
  <div
    className="ui padded segment"
  >
    No configuration values are provided
  </div>
  <div
    className="mt3 clearfix"
  >
    <button
      className="ui blue icon disabled right floated left labeled button ml2"
      disabled={true}
      onClick={[Function]}
      tabIndex={-1}
    >
      <i
        aria-hidden="true"
        className="save icon"
        onClick={[Function]}
      />
      Save
    </button>
    <button
      className="ui icon right floated left labeled button"
      onClick={[Function]}
    >
      Previous
      <i
        aria-hidden="true"
        className="left arrow icon"
        onClick={[Function]}
      />
    </button>
    <button
      className="ui left floated button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;
